modules/agent/pkg/deployer/monitor.go

```diff
@@ -288,5 +297,51 @@ func nonReady(plan apply.Plan) (result []fleet.NonReadyStatus) {
 		}
 	}
 
-	return
+	return result
+}
+
+// excludeIgnoredConditions remove the conditions that are included in ignoreOptions from the object passed as a parameter
+func excludeIgnoredConditions(obj *unstructured.Unstructured, ignoreOptions fleet.IgnoreOptions) error {
+	conditions, _, err := unstructured.NestedSlice(obj.Object, "status", "conditions")
+	if err != nil {
+		return err
+	}
+	conditionsWithoutIgnored := make([]interface{}, 0)
+
+	for _, condition := range conditions {
+		condition, ok := condition.(map[string]interface{})
+		if !ok {
+			return fmt.Errorf("condition: %#v can't be converted to map[string]interface{}", condition)
+		}
+		excludeCondition := false
+		for _, ignoredCondition := range ignoreOptions.Conditions {
+			if shouldExcludeCondition(condition, ignoredCondition) {
+				excludeCondition = true
+				break
+			}
+		}
+		if !excludeCondition {
+			conditionsWithoutIgnored = append(conditionsWithoutIgnored, condition)
+		}
+	}
+
+	err = unstructured.SetNestedSlice(obj.Object, conditionsWithoutIgnored, "status", "conditions")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// shouldExcludeCondition returns true if all the elements of ignoredConditions are inside conditions
```

Comment:
I feel that this is an aspect of this new behaviour that we'll want to mention in user-facing docs ;)

Reply:
https://github.com/rancher/fleet-docs/pull/62